gabi2/src/main/kotlin/factoryPattern/CarFactory/BenzCarFactory.kt

```diff
@@ -0,0 +1,16 @@
+package factoryPattern.CarFactory
+
+import factoryPattern.Car.Car
+import factoryPattern.Car.SUV
+import factoryPattern.Car.Sedan
+
+class BenzCarFactory : CarFactory() {
+    override fun createCar(type: String): Car? {
+        val car = when (type) {
+            "SUV" -> SUV()
+            "Sedan" -> Sedan()
+            else -> null
+        }
+        return car
+    }
+}
```

Comment:
```suggestion
enum CarType{
    SUV,
    SEDAN;
}
class BenzCarFactory : CarFactory() {
    override fun createCar(type: CarType): Car {
        return when (type) {
            CarType.SUV -> SUV()
            CarType.SEDAN -> Sedan()
        }
    }
}
```

대충 짜서 틀렸을 수도 있는데 이렇게 하면 nullable 하지 않게 짤수 있을것 같아요